### PR TITLE
fix: [IOPID-2490] Add  correct check on whitelisted URLs

### PR DESCRIPTION
**Pull request by Ladirico**
> [!Note]
> This PR is related to this PR https://github.com/pagopa/io-app/pull/6421

## Short description
As suggested by the security team in this PR, a more robust check has been added that checks the return url, checking both the protocol and hostname to see if it is whitelisted

## Demo

<p>

| iOS unhappy and happy path | Android unhappy path (goldfish) | Android unhappy path (evil url) | Android happy path | 
| - | - | - | - |
| <video src="https://github.com/user-attachments/assets/b16f6181-1610-4db1-8220-46e2149317c7"/> | <video src="https://github.com/user-attachments/assets/2519cc9b-c7ab-4abc-ab8e-fcbd0ea91894"/> | <video src="https://github.com/user-attachments/assets/4124ee45-4c4f-48ad-9ebf-3434a6da65bb"/> | <video src="https://github.com/user-attachments/assets/d6f1989e-7b3a-4136-adf9-1325793a0355"/> |

</p>

## How to test UNHAPPY PATH
- run the application using the production environment
- if you're using an iOS follow the video and attach in safari this urls: 
  - `iologincie:https://idserver.servizicie.interno.gov.it.evilhacker.org `
  - `iologincie:https://it.wikipegia.org/wiki/File:Goldfish3.jpg`
- if you're using an Android device you need to install on device a fake cieid app (you can send me a message and i can help you to create and install this application)

## How to test HAPPY PATH
- install CieID app
- run IO application using the production environment
- test as you can see in Demo section

### CODE REVIEW

File: ts/features/cieLogin/components/CieIdLoginWebView.tsx [hunk 1]
```diff
@@ -113,14 +114,22 @@ const CieIdLoginWebView = ({ spidLevel, isUat }: CieIdLoginProps) => {
...
-        // Set the URL as valid
-        setAuthenticatedUrl(url);
-      } else {
+
+      try {
+        const { hostname, protocol } = new URL(url);
+
+        const isDomainValid = WHITELISTED_DOMAINS.includes(
+          `${protocol}//${hostname}`
+        );
```
Comment (ChrisMattew) [inline ts/features/cieLogin/components/CieIdLoginWebView.tsx:123]:
```suggestion
        const { origin } = new URL(url);

        const isDomainValid = WHITELISTED_DOMAINS.includes(origin);
```
Have you tried with `origin` parameter? Could it work?

**Review by shadowsheep1 — COMMENTED** [review]
LGTM

**Review by ChrisMattew — APPROVED** [review]
LGTM!

File: ts/features/cieLogin/components/CieIdLoginWebView.tsx (hunk 1 above)
Comment (Ladirico) [reply ts/features/cieLogin/components/CieIdLoginWebView.tsx]: yesss done! Good suggestion!

**Pull request merged**